recipes/jsonnet/all/patches/0.20.0/0003-use-cpp17.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index c14c93c..b1cb1b3 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -43,7 +43,7 @@ else()
+     message(WARNING "Compiler ${CMAKE_CXX_COMPILER_ID} not supported")
+ endif()
+ 
+-set(CMAKE_CXX_STANDARD 11)
```

Comment:
Not sure that just changing (and hardcoding the C++ standard) is the best way. I guess that if users define ``compiler.cppstd=20`` it will be mostly ignored and overwritten because of this being hardcoded here?

(not saying this is a problem of this PR, but it seems the problem was there in the past too)

Reply:
@memsharded 
Thank you for your advise.
You are right. It is not mandatory.
I will remove it.

It is an upstream patch which have been merged already.
https://github.com/google/jsonnet/pull/1076

Reply:
@memsharded 
I met a compilation error caused by `set(CMAKE_CXX_STANDARD 11)` in msvc.
May I apply `0003-use-cpp17.patch` again?


```
error C2429: language feature 'nested-namespace-definition' requires compiler flag '/std:c++17' 
(compiling source file C:\J\w\prod\BuildSingleReference@2\.conan\data\jsonnet\0.20.0\_\_\build
```

Reply:
My concern here is that it passes CI, but it is causing a mis-alignment of the resulting binaries. No matter what users pass ``compiler.cppstd=20`` or ``compiler.cppstd=14``, that apparently should be fine in some compilers, but failing in others. And the change affect all compilers (but not changing it is also not great, because it is still misaligned with the provided ``compiler.cppstd``

The ideal (for every package, not only this) would be:
- Remove it from CMakeLists.txt
- The ``conan_toolchain.cmake`` will define it, and it is mandatory in Conan 2.0
- If Conan 2.0-only support was possible we would be done. But if we need to maintain the 1.X pipeline and behavior, we could explicitly define it when ``compiler.cppstd=None`` in the ``toolchain.cache_variables["CMAKE_CXX_STANDARD"]`` to the value that it needs.

I am approving this PR, this is a net improvement, it only affects the new version, this can be addressed later.